packages/api/src/user.js

```diff
@@ -357,36 +406,39 @@ export async function userPinsGet (request, env) {
   try {
     pinRequests = await env.db.listPsaPinRequests(tokens, {
       ...psaParams.data,
-      limit: size,
-      offset,
-      before,
-      after,
-      sortBy,
-      sortOrder
+      limit: pageRequest.size,
+      offset: pageRequest.size * (pageRequest.page - 1)
     })
-  } catch (e) {
-    console.error(e)
-    throw new HTTPError('No pinning resources found for user', 404)
+  } catch (err) {
+    if (err.code === 'RANGE_NOT_SATISFIABLE_ERROR_DB') {
```

Comment:
Maybe reuse RangeNotSatisfiableDBError.CODE here

Reply:
Great point! I could make a constant in the API package so that there's less dependancy on the `db` package, or import it over from the db package. am struggling to weight up which would be best!